app/furniture/marketplace/square_order.rb

```diff
@@ -0,0 +1,167 @@
+# SquareOrder encapsulates logic for coordinating and executing interactions
+# between a Convene Marketplace and the seller's Square account. It could be
+# described as a (budding) service or process class, but current behavior is
+# limited to only registering orders in a seller's account upon sale.
+#
+# Registering orders serves to both synchronize record keeping between
+# Marketplace managers owners (eg. Pikkkup) and sellers (eg. Crumble & Wisk)
+# and automatically notify sellers of new orders.
+#
+# NOTE: the simple `Markeplace > SquareOrder` namespacing was chosen
+# to avoid collision with the `Square` namespace already included by Square's gem.
+#
+# TODOS/considerations:
+# 1. Implement validation of `order` argument on initialization
+#   Eg: if order.is_not_valid?; exit; end
+# 2. Implement state tracking
+#   Eg: READY | ORDER_SENT | etc...
+class Marketplace
+  class SquareOrder
+    def initialize(order)
+      @order = order
+      @marketplace = order.marketplace
+      @shopper = order.shopper
+      @space = order.marketplace.space
+      @ordered_products = @order.ordered_products
+      @square_location_id = @marketplace.settings["square_location_id"]
+      @square_order_id = nil
+      @square_payment_id = nil
+      # @state = "READY"
+    end
+
+    # Sends an order to Square. Will create both an order and payment in the
+    # receiving account which are requirements for the order to show up in the
+    # Seller Dashboard.
+    def send_order
+      Rails.logger.info("Start creating Square order")
```

Comment:
Ah ok. A blend of misinterpreting your guidance and forgetting we had an event system implemented. Let's do it.

Reply:
@zspencer 

What's the thinking around what strings go into the `public.marketplace_events.description` column since it's a "free" `text` type?  Understandably with a small org code review serves as a guardrail to maintain consistent event descriptions/actions. But would the list of accepted descriptions strings be something we would consider declaring in some kind of _enum_ somewhere so developers could easily consult an existing set? 

I already notice we use a typical pattern of Title Case, past tense, and Noun + Verb. What about adding a small style guide comment to the `Event` model?

Reply:
I think a style guide is probably better than an enum; but if you want to gather the usages I wouldn't be opposed.